wp-content/plugins/inn-misc-functionality/inc/shortcode_innovation_mailchimp.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * INNovation mailchimp newsletter shortcodes
+ *
+ * @link https://github.com/INN/umbrella-inndev/issues/71
+ * @since 0.2
+ */
+
+namespace INN\Plugin\INNovation;
+
+/**
+ * The shortcode for the INNovation Newsletter archive
+ *
+ * @param array $atts unused
+ * @param string $content unused
+ * @param string $tag unused
+ * @since 0.2
+ */
+function innovation_newsletter_archive( $atts, $content, $tag ) {
+	$return = <<<'EOT'
```

Comment:
Should `$return` here and on line 37 be named something more specific?

Reply:
Is there something it could be named that's more specific?

If we could do this, I'd prefer this formulation, but it didn't work the last time I tried:

```php
function foo() {
    return <<<'EOT'
blah blah
EOT;
}

Reply:
Maybe `$archive_script` and `$newsletter_form`? You're right, I don't think returning an EOT not inside of a variable works.